crd/apis/network/v1/gkenetworkparams_types.go

```diff
@@ -0,0 +1,48 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:storageversion
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GKENetworkParams represent GKE specific parameters for the network.
+type GKENetworkParams struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   GKENetworkParamsSpec   `json:"spec,omitempty"`
+	Status GKENetworkParamsStatus `json:"status,omitempty"`
+}
+
+// GKENetworkParamsSpec contains the specifications for network object
+type GKENetworkParamsSpec struct {
+	// VPC speficies the VPC to which the network belongs.
+	// +required
+	VPC string `json:"vpc"`
+
+	// VPCSubnet is the path of the VPC subnet
+	// +required
+	VPCSubnet string `json:"vpcSubnet"`
+
+	// PodIPv4Ranges specify the names of the secondary ranges of the VPC subnet
+	// used to allocate pod IPs for the network
+	// +optional
+	PodIPv4Ranges []string `json:"podIPv4Ranges,omitempty"`
```

Comment:
This should be a pointer to struct that has the list of strings. That is how we have defined the CRD in our design. In any case, it should be consistent with the structure of PodCIDRs in the status.